packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -417,15 +425,16 @@ function registerBuilderComponents(): ThunkAction<() => void, State, unknown, Ac
     const componentsMeta = getComponentsMeta(state)
 
     componentsMeta.forEach((meta, type) => {
-      const propControllerDescriptors = getComponentPropControllerDescriptors(state, type)
-      if (propControllerDescriptors != null) {
-        dispatch(registerComponent(type, meta, propControllerDescriptors))
```

Comment:
We were repurposing `registerComponent` here to effectively dispatch `registerBuilderComponent`. This was confusing, replaced with a proper `registerBuilderComponent` dispatch.